### ISSUE TITLE
Update Connection Strings on Production slot returns Bad Request

### ISSUE THREAD

**Issue by simone-cw** [Azure/azure-rest-api-specs#20499]
We encountered an error when attempting to update Connection Strings on a Production slot for any of our App Services.
The issue was first noticed when running the Set-AzWebApp command which uses the API described below

**API Endpoint** - [Web Apps - Update Connection Strings
](https://docs.microsoft.com/en-us/rest/api/appservice/web-apps/update-connection-strings)
`https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/config/connectionstrings?api-version=2022-03-01`

**Verb** - PUT

**Request**
```
PUT - https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/config/connectionstrings/list?api-version=2020-12-01
{
  "id": "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/config/connectionstrings",
  "name": "connectionstrings",
  "type": "Microsoft.Web/sites/config",
  "location": "{region",
  "tags": {
    "provider": "*****",
    "logicalName": "cd"
  },
  "properties": {
    "{dbname}": {
      "value": "******",
      "type": "SQLAzure"
    }
  }
}
```

**Response**
![image](https://user-images.githubusercontent.com/103508091/187687869-25392aa7-4e73-436a-aee7-a13c671daab7.png)

- The issue is also reproducible through the [Azure Resource Explorer](https://resources.azure.com/).
- In our tests we've noticed that the similar API [Web Apps - Update Connection Strings Slot](https://docs.microsoft.com/en-us/rest/api/appservice/web-apps/update-connection-strings-slot) works without any issues
- The issue is reproducible when either passing one, multiple or no connection strings at all.
- The issue is not reproducible through the Azure Portal since it relies on a different API when updating the App Service configuration
- The issue is replicable in our current tenant/subscription, however we were not able to reproduce it in other tenants.

**Comment by simone-cw** [comment]
We've managed to find the root cause of the issue, specifically an Azure policy that was intercepting the API call and returning a Bad Request before even reaching the [Web Apps - Update Connection Strings](https://docs.microsoft.com/en-us/rest/api/appservice/web-apps/update-connection-strings) API.

We've opened a feature request with Azure to provide a more helpful error message as there was no indication of the Bad Request response originating from the Azure Policy. 

Happy to have this issue closed.
